src/test/iot/commands/attachPolicy.test.ts

```diff
@@ -22,11 +23,17 @@ describe('attachPolicyCommand', function () {
     let certNode: IotCertWithPoliciesNode
     let window: FakeWindow
     let selection: number = 0
-    const prompt: (policyItems: DataQuickPickItem<IotPolicy>[]) => Promise<PromptResult<IotPolicy>> = policyItems => {
-        return new Promise((resolve, reject) => {
-            resolve(selection > -1 ? (policyItems[selection].data as IotPolicy) : undefined)
-        })
-    }
+    const prompt: (iot: IotClient, policyFetch: policyGen, window?: Window) => Promise<PromptResult<Iot.Policy>> =
+        async (iot, policyFetch) => {
+            const iterable = policyFetch(iot, window)
+            const responses: DataQuickPickItem<Iot.Policy>[] = []
+            for await (const response of iterable) {
+                responses.push(...response)
+            }
+            return new Promise((resolve, reject) => {
+                resolve(selection > -1 ? (responses[selection].data as Iot.Policy) : undefined)
+            })
```

Comment:
```suggestion
return selection > -1 ? (responses[selection].data as Iot.Policy) : undefined
```
Creating the promise isn't needed anymore. We've marked the function `async` so it will always return a Promise.